metadata_backend/helpers/metax_mapper.py

```diff
@@ -0,0 +1,560 @@
+                                "type": {
+                                    "description": "Type of the entity, for example: API, Application, News article, "
+                                    "paper, post or visualization.",
+                                    "type": "object",
+                                    "$ref": "#/definitions/Concept",
+                                },
+                            },
+                        },
+                        "relation_type": {
+                            "description": "Role of the influence.",
+                            "type": "object",
+                            "$ref": "#/definitions/Concept",
+                        },
+                    },
+                },
+            },
+            # subject Yrjö Leino
+            "field_of_science": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/Concept"},
+            },
+            # TODO: Needs clarification
+            "remote_resources": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/WebResource"},
+            },
+            # restricted - ask Metax team for a link to REMS
+            "access_rights": {
+                "type": "object",
+                "$ref": "#/definitions/RightsStatement",
+            },
+            # contributors - Data Curator
+            "curator": {
+                "type": "array",
+                "items": {"type": "object", "$ref": "#/definitions/ResearchAgent"},
+            },
+            # sizes field need either some indication on incomming type of data or MUST be in bytes
+            # Though now dates schema is array so maybe data type is better solution for multiple object inputs
+            "total_remote_resources_byte_size": {
+                "type": "integer",
+            },
+        }
+    }
+
+    def __init__(self, metax_data: Dict, data: Dict) -> None:
+        """Set variables.
+
+        :param metax_data: Metax research_dataset metadata
+        """
+        self.research_dataset = metax_data
+        self.datacite_data = data
+        self.identifier_types = metax_reference_data["identifier_types"]
+        self.person: Dict[str, Any] = {
+            "name": "",
+            "@type": "Person",
+            "member_of": {"name": {"en": ""}, "@type": "Organization"},
+            "identifier": "",
+        }
+
+    def map_metadata(self) -> Dict[str, Any]:
+        """Public class for actual mapping of metadata fields.
+
+        :returns: Research dataset
+        """
+        LOG.info("Mapping datasite data to Metax metadata")
+        LOG.debug("Data incomming for mapping: ", self.datacite_data)
+        for key, value in self.datacite_data.items():
+            if key == "creators":
+                self._map_creators(value)
+            if key == "keywords":
+                self.research_dataset["keyword"] = value
```

Comment:
with the new changes we might need to split the value by comma to make this an array field as required by metax API.